docs/concepts/overview/working-with-objects/common-labels-annotations.md

```diff
@@ -0,0 +1,186 @@
+---
+title: Recommended Labels and Annotations
+---
+
+{% capture overview %}
+Many tools can be used to visualize and manage Kubernetes objects. The tools used
+to manage objects goes beyond kubectl and the dashboard. To enable interoperability
+between tools, a common set of labels and annotation allows objects to be described
+in a common manner that all tools can understand.
+
+In addition to supporting tooling, the recommended labels and annotations describe
+applications, in a generic sense, in a way that can queried or are otherwise useful
+when operating applications.
+{% endcapture %}
+
+{% capture body %}
+The recommended object metadata is broken apart so that some of it is labels while
+other parts are annotations. Labels are shorter and can be used when querying
+the Kubernetes API. Annotations allow for longer data that's not meant to be
+queried.
+
+The metadata is organized around the concept of an Application. Kubernetes is not
+a Platform as a Service and doesn't have or enforce a formal notion of an application.
+Instead, applications are informal, described with metadata, and the definition of what an
+application contains is loose.
+
+Note, these are recommended labels and annotations. They aid the use of managing
+applications but are not required for any core tooling to work.
+
+One common theme you'll notice about the labels and annotations is that they
+contain a prefix of `app.kubernetes.io`. Labels and annotations without a prefix
+are considered private to users. These are meant to be common labels that do not
+interfere with existing or custom user labels. To accomplish this a prefix is
+used.
+
+## Labels
+
+| Key                                 | Description           | Example  | Type |
+| ----------------------------------- | --------------------- | -------- | ---- |
+| `app.kubernetes.io/name`              | The name of the application stack | `wordpress` | string |
+| `app.kubernetes.io/version`           | The current version of the running application (e.g., a semantic version, revision hash, etc.) | `v1.2.3` | string |
+| `app.kubernetes.io/component`         | The component within the architecture | `front-end` | string |
+| `app.kubernetes.io/component-name`    | The name of a component within the application | `mysql` | string |
+| `app.kubernetes.io/component-version` | The current version of the component | `5.7.21` | string |
+| `app.kubernetes.io/deploy-manager`    | The tool being used to manage the operation of an application | `helm` | string |
```

Comment:
First, this is a label because you may want to query on it. For example, if you are migrating from one tool to another you may want to query for all the objects being managed by each tool. Or, a tool may way to query for all the objects it's managing. There are querying use cases and the names are short enough they can fit into a label.

Second, you asked about a tool deploying via Helm. I would suggest you look at Draft. Helm has a set of recommended labels today. The closest relation to this property is one called heritage. Draft sets the heritage to draft since it's the tool at the top doing the managing.

Reply:
How does this relate to objects created by controllers - e.g. a ReplicaSet controller create a Pod.  Does the Pod have a deployment manager label?  Is it the replicaset-controller itself or the deployment-manager of the instance replicaset?

What does this look like for deployments managed by a composition of tools?  e.g. if you applied an instantiated helm template would this still be helm, apply, both, neither?

Reply:
Is this also being discussed in the sig-architecture meeting?

Reply:
`deploy-manager`: I wouldn't think so. That's not a project-spanning issue.

Reply:
I guess it sort of is since the choice of label vs. annotation would affect what type of applications can query on the values and how they would cooperate in the preservation of data (via appending or last-tool-wins), but it's definitely less cross-project than prefixing.

Reply:
Ah. I still contend that this shouldn't be a label. It isn't an identifying attribute.

Reply:
+1 on annotation.